### PR TITLE
Project Membership by semester - API work

### PR DESCRIPTION
**Pull request by jessamarie**
This PR intends to add functionality for the Project Members Project.

So far:
- Modified the Projects object in the user model to hold the project AND the semesters on the project
- Modified the api user controller:
  - removeProject removes entire project only if there is the current semester in the array OR just     
   removes the current semester if the user has multiple semesters on the project.
  - add project adds an entire project if it doesn't exist in the user's array OR just adds the current   
   semester on the project
- I have not yet done any client side modifications. 

Any tips on how to test this? Postman , maybe?

Same as #623

### CODE REVIEW

File: server/config/seed/users.json [hunk 1]
```diff
@@ -2,6 +2,12 @@
     "provider": "local",
     "name": "Test User",
     "email": "test@test.com",
+    "projects": [{
+      "project": "000000000000000000000010",
+       "classYears": [{"id":"000000000000000000000001",
+                        "joinDate": "this is totally a date string"}],
```
Comment (kmcnellis) [inline server/config/seed/users.json]: You should replace this with an actual date!

File: server/config/seed/users.json [hunk 2]
```diff
@@ -2,6 +2,12 @@
     "provider": "local",
     "name": "Test User",
     "email": "test@test.com",
+    "projects": [{
+      "project": "000000000000000000000010",
+       "classYears": [{"id":"000000000000000000000001",
```
Comment (kmcnellis) [inline server/config/seed/users.json]: Lets keep this consistent, and change classYears.id into classYears. classYear

Comment (kmcnellis) [inline server/config/seed/users.json]: Add a newline after the {

File: server/config/seed/users.json (hunk 1 above)
Comment (kmcnellis) [inline server/config/seed/users.json]: Add a newline before the }

Comment (jessamarie) [inline server/config/seed/users.json]: Will do!

**Comment by jessamarie** [comment]
I will try to make these changes and also start working on the migration script by Tuesday's small group session!

**Comment by kmcnellis** [comment]
Merging into the feature branch!

**Pull request merged**